new-log-viewer/src/services/LogFileManager.ts

```diff
@@ -0,0 +1,138 @@
+import {DecodeOptionsType} from "../typings/decoders";
+import {
+    CursorType,
+    FileSrcType,
+    LineNumLogEventNumMap,
+} from "../typings/worker";
+import {getUint8ArrayFrom} from "../utils/http";
+import {getBasenameFromUrl} from "../utils/url";
+import {
+    Decoder,
+    DecoderConstructor,
+} from "./decoders/Decoder";
+import JsonlDecoder from "./decoders/JsonlDecoder";
+
+
+/**
+ * A mapping of file name extensions to their respective decoders.
+ */
+const FILE_EXT_TO_DECODER: Readonly<Record<string, DecoderConstructor>> = Object.freeze({
+    ".jsonl": JsonlDecoder as unknown as DecoderConstructor,
```

Comment:
Rather than breaking out of the type system, could we just handle all the cases in the `loadFile`? Is there anywhere else where we would need to determine the decoderconstructor at runtime?

Reply:
Not sure if I understood the question correctly. The goal of storing a map of file extensions to decoder constructors is to allow different types of decoders to be created depending on the file name extension. Note:
1. The method signature of some class's constructor must be separate from the class's definition, as done in Decoder.ts.
2. Eventually we will provide a magic number based file type detection mechanism.

Reply:
Discussed offline and we concluded that inlining the cases into the code would be better.